radp/digital_twin/mobility/ue_generation_handler.py

```diff
@@ -0,0 +1,257 @@
+
+
+    def plot_ue_tracks(self,csv_file) -> None:
+
+        """
+        Plot UE tracks from a CSV file in respect to time. 
+        Each plotted graph represents a batch of data, and the lines and arrows in the graph
+        are used to represent the direction of movement of the respective UE's.
+        
+        param csv_file: Path to the CSV file containing the UE Tracks data.
+        """
+        
+        # Load the data
+        data = pd.read_csv(csv_file)
+
+        # Initialize an empty list to store batch indices
+        batch_indices = []
+
+        # Identify where tick resets and mark the indices
+        for i in range(1, len(data)):
+            if data.loc[i, 'tick'] == 0 and data.loc[i-1, 'tick'] != 0:
+                batch_indices.append(i)
+
+        # Add the final index to close the last batch
+        batch_indices.append(len(data))
+
+        # Now, iterate over the identified batches
+        start_idx = 0
+        for batch_num, end_idx in enumerate(batch_indices):
+            batch_data = data.iloc[start_idx:end_idx]
+
+            # Create a new figure
+            plt.figure(figsize=(10, 6))
+
+            # Generate a color map with different colors for each ue_id
+            color_map = cm.get_cmap('tab20', len(batch_data['mock_ue_id'].unique()))
+
+            # Plot each ue_id's movement over ticks in this batch
+            for idx, ue_id in enumerate(batch_data['mock_ue_id'].unique()):
+                ue_data = batch_data[batch_data['mock_ue_id'] == ue_id]
+                color = color_map(idx)  # Get a unique color for each ue_id
+
+                # Plot the path with arrows
+                for i in range(len(ue_data) - 1):
+                    x_start = ue_data.iloc[i]['longitude']
+                    y_start = ue_data.iloc[i]['latitude']
+                    x_end = ue_data.iloc[i+1]['longitude']
+                    y_end = ue_data.iloc[i+1]['latitude']
+
+                    # Calculate the direction vector
+                    dx = x_end - x_start
+                    dy = y_end - y_start
+
+                    # Plot the line with an arrow with reduced width and unique color
+                    plt.quiver(x_start, y_start, dx, dy, angles='xy', scale_units='xy', scale=1, color=color,
+                              width=0.002, headwidth=3, headlength=5)
+
+                # Plot starting points as circles with the same color
+                plt.scatter(ue_data['longitude'].iloc[0], ue_data['latitude'].iloc[0], color=color, label=f'Start UE {ue_id}')
+
+            # Set plot title and labels
+            plt.title(f'UE Tracks with Direction for Batch {batch_num + 1}')
+            plt.xlabel('Longitude')
+            plt.ylabel('Latitude')
+            plt.legend()
+
+            # Display the plot
+            plt.show()
+
+            # Update start_idx for the next batch
+            start_idx = end_idx
+
+    def save_data_to_csv(self, filename: str) -> None:
```

Comment:
you may delete this (in future, you may provide such a utility if we need to build in more business logic around this export)